src/util/encode-url-for-id.js

```diff
@@ -22,21 +22,21 @@ const normalizationOpts = {
  */
 function applyQueryParamsRules(url) {
     const parsed = new URL(url)
-    const rules = queryParamRules.get(parsed.hostname)
+    const rulesObj = queryParamRules.get(parsed.hostname)
 
     // Base case; domain doesn't have any special normalization rules
-    if (!rules) {
+    if (!rulesObj) {
         return url
     }
 
     // Remove all query params that don't appear in special rules
-    const rulesSet = new Set(rules)
-    for (const param of parsed.searchParams.keys()) {
-        if (!rulesSet.has(param)) {
-            parsed.searchParams.delete(param)
-        }
-    }
-
+    const rulesSet = new Set(rulesObj.rules)
+    const rulesType = rulesObj.type
+	for (const param of parsed.searchParams.keys()) {
+		if ((rulesType === 'keep' && !rulesSet.has(param)) || (rulesType == 'remove' && rulesSet.has(param))) {
```

Comment:
Looks better, but I think the condition would be more readable and less chance of unintended logical errors by putting its evaluation on a separate line to avoid all the "and"s and "or"s, like:
```js
const shouldDelete = rulesType === 'keep' ? !rulesSet.has(param) : rulesSet.has(param)
if (shouldDelete) { ... }
```

Another thing: is your dev environment set up properly (i.e., you have run `yarn` to install all deps)? Asking as this line should have been cut to multiple lines automatically by `prettier` on pre-commit event. The linter should also be giving you warnings for the abstract equality op and missing trailing comma in L132 in the rules file. May also be an error in our dev setup, so send me a message if you're unsure and we can get it set up 👍

Reply:
Yeah it's all set up, I just forgot to run it this one time 😅